### PR TITLE
Team Assistant Feature

### PR DESCRIPTION
**Pull request by tsa87**
WIP

- Display nested macros in Flutter
- Create new notifier and util functions in Dart
- Upcoming Server additions...

### CODE REVIEW

File: flutter_src/lib/logic/api/firestore_db.dart [hunk 1]
```diff
@@ -1,32 +1,58 @@
...
+        Team team = TeamModel.fromJson(element.data);
+        team.teamId = element.documentID;
+        _teamList.add(team);
+      } catch(e) {
+        print(element.data);
+      }
+    });
+
+    teamNotifier.teamList = _teamList;
+  }
```
Comment (tsa87) [inline flutter_src/lib/logic/api/firestore_db.dart]: These two functions look really similar, however, I wasn't sure how to combine them into one generic function.

File: flutter_src/lib/logic/state/team_state/team_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import 'dart:collection';
+import 'package:flutter/material.dart';
+import 'package:macrobaseapp/model/entities/team.dart';
+
+class TeamNotifier with ChangeNotifier {
```
Comment (tsa87) [inline flutter_src/lib/logic/state/team_state/team_notifier.dart:5]: Also too similar to `macro_notifier.dart`

File: flutter_src/lib/logic/api/firestore_db.dart (hunk 1 above)
Comment (lipczak-g) [reply flutter_src/lib/logic/api/firestore_db.dart]: yeah, I can see the structural similarity, but the types make them fairly different, so I think they are better to be kept separated. BTW, you fixed the style in the top onem but not the bottom one.

Comment (lipczak-g) [reply flutter_src/lib/logic/api/firestore_db.dart]: Just FYI: In Java you would use generics to do it, but again, not worth the effort here... too simple
https://docs.oracle.com/javase/tutorial/extra/generics/methods.html

File: flutter_src/lib/logic/state/team_state/team_notifier.dart (hunk 2 above)
Comment (lipczak-g) [reply flutter_src/lib/logic/state/team_state/team_notifier.dart:5]: Again generic class could be a solution here, but it is not worth here. Sometimes clarity is more important than concise code.

File: flutter_src/lib/model/entities/action.dart [hunk 3]
```diff
@@ -67,7 +67,11 @@ class BatchAction extends SheetAction {
 }
 
 class AppendAction extends SheetAction {
-  static const List<String> VALUE_LIST = ["EMPTY", "TIME", "EMAIL", "CONTENT"];
+  static const List<String> VALUE_LIST = [VALUE_EMPTY, VALUE_TIME, VALUE_EMAIL, VALUE_CONTENT];
+  static const VALUE_EMPTY = "EMPTY";
```
Comment (lipczak-g) [inline flutter_src/lib/model/entities/action.dart:71]: wonder how this works when VALUE_* consts are defined after VALUE_LIST

File: flutter_src/lib/logic/api/firestore_db.dart (hunk 1 above)
Comment (tsa87) [reply flutter_src/lib/logic/api/firestore_db.dart]: Thanks for letting me know about Generics! Yea, in this case it seems okay to keep it this way for now.

File: flutter_src/lib/model/entities/action.dart (hunk 3 above)
Comment (tsa87) [reply flutter_src/lib/model/entities/action.dart:71]: Changed the order

**Pull request merged**